File: starter_kits/CSharp/Halite3/hlt/Dropoff.cs
namespace Halite3.hlt
{
    /// <summary>
    /// A Dropoff is a type of Entity where allied ships can drop off halite.
    /// </summary>
    /// <see cref="https://halite.io/learn-programming-challenge/api-docs#dropoff"/>
    public class Dropoff : Entity
    {
        public Dropoff(PlayerId owner, EntityId id, Position position) : base(owner, id, position)
        {
        }

        /// <summary>
        /// Creates a Dropoff instance based on engine output.
        /// </summary>
        public static Dropoff _generate(PlayerId playerId)
        {
            Input input = Input.ReadInput();

            EntityId dropoffId = new EntityId(input.GetInt());
            int x = input.GetInt();
            int y = input.GetInt();

            return new Dropoff(playerId, dropoffId, new Position(x, y));
        }
    }
}

File: starter_kits/Clojure/hlt/lib/hlt/dropoff.clj
(ns hlt.dropoff
  (:require [hlt.input :refer [read-line-ints]]
            [hlt.position :refer [new-position]]))

(defn id
  "Get id for a dropoff"
  [dropoff]
  (::id dropoff))

(defn owner
  "Get owner for a dropoff"
  [dropoff]
  (:hlt.player/id dropoff))

(defn position
  "Get position of a dropoff"
  [dropoff]
  (:position dropoff))

(defn new-dropoff
  "New dropoff"
  [player-id dropoff-id position]
  {::id dropoff-id
   :hlt.player/id player-id
   :position position})

(defn read-dropoff
  "Read dropoff"
  [player-id]
  (let [[dropoff-id x y] (read-line-ints)
        position (new-position x y)
        dropoff (new-dropoff player-id dropoff-id position)]
    [dropoff-id dropoff]))
